src/vs/workbench/contrib/terminal/common/terminalConfiguration.ts

```diff
@@ -142,7 +142,55 @@ export const terminalConfiguration: IConfigurationNode = {
 				comment: ['{0}, {1}, and {2} are the `path`, `profileName`, and optional `args` settings keys']
 			},
 				"The osx shell profiles to select from when creating a new terminal via the terminal dropdown. When set, these will override the default detected profiles. They are comprised of a {0}, {1}, and optional {2}", '`path`', '`profileName`', '`args`.'),
-			type: 'array'
+			type: 'object',
+			properties: {
+				'gitbash': {
+					markdownDescription: 'gitbash',
+					type: ['object', 'null'],
```

Comment:
I haven't done this yet, but it looks like `patternProperties` might be the way you're mean to validate `type: 'object'`

Reply:
Deferred: https://github.com/microsoft/vscode/issues/118998